tests/test_alignment.py

```diff
@@ -0,0 +1,182 @@
+    wcs = _create_wcs_object_without_distortion(
+        fiducial_world=fiducial_world, shape=shape, pscale=pscale
+    )
+    if datamodel_type == "jwst":
+        datamodel = _create_jwst_meta(shape, fiducial_world, wcs)
+    elif datamodel_type == "roman":
+        datamodel = _create_roman_meta(shape, fiducial_world, wcs)
+
+    return datamodel
+
+
+def _create_jwst_meta(shape, fiducial_world, wcs):
+    result = ImageModel(np.zeros(tuple(shape.value.astype(int))))
+
+    result.meta.wcsinfo.ra_ref = fiducial_world[0].value
+    result.meta.wcsinfo.dec_ref = fiducial_world[1].value
+    result.meta.wcsinfo.ctype1 = "RA---TAN"
+    result.meta.wcsinfo.ctype2 = "DEC--TAN"
+    result.meta.wcsinfo.v2_ref = 0
+    result.meta.wcsinfo.v3_ref = 0
+    result.meta.wcsinfo.roll_ref = 0
+    result.meta.wcsinfo.v3yangle = 0
+    result.meta.wcsinfo.vparity = -1
+    result.meta.wcsinfo.wcsaxes = 2
+
+    result.meta.coordinates.reference_frame = "ICRS"
+
+    result.meta.wcs = wcs
+
+    return result
+
+
+def _create_roman_meta(shape, fiducial_world, wcs):
+    result = utils.mk_level2_image(shape=tuple(shape.value.astype(int)))
+
+    result.meta.wcsinfo.ra_ref = fiducial_world[0].value
+    result.meta.wcsinfo.dec_ref = fiducial_world[1].value
+    result.meta.wcsinfo.v2_ref = 0
+    result.meta.wcsinfo.v3_ref = 0
+    result.meta.wcsinfo.roll_ref = 0
+    result.meta.wcsinfo.v3yangle = 0
+    result.meta.wcsinfo.vparity = -1
+
+    result.meta.coordinates.reference_frame = "ICRS"
+
+    result.meta["wcs"] = wcs
+
+    result = rdm.ImageModel(result)
+
+    return result
+
+
+def test_compute_fiducial():
+    """Test that util.compute_fiducial can properly determine the center of the
+    WCS's footprint.
+    """
+
+    shape = (3, 3) * u.pix
+    fiducial_world = (0, 0) * u.deg
+    pscale = (0.05, 0.05) * u.arcsec
+
+    wcs = _create_wcs_object_without_distortion(
+        fiducial_world=fiducial_world, shape=shape, pscale=pscale
+    )
+
+    computed_fiducial = compute_fiducial([wcs])
+
+    assert all(np.isclose(wcs(1, 1), computed_fiducial))
+
+
+@pytest.mark.parametrize("pscales", [(0.05, 0.05), (0.1, 0.05)])
+def test_compute_scale(pscales):
+    """Test that util.compute_scale can properly determine the pixel scale of a
+    WCS object.
+    """
+    shape = (3, 3) * u.pix
+    fiducial_world = (0, 0) * u.deg
+    pscale = (pscales[0], pscales[1]) * u.arcsec
+
+    wcs = _create_wcs_object_without_distortion(
+        fiducial_world=fiducial_world, shape=shape, pscale=pscale
+    )
+    expected_scale = np.sqrt(pscale[0].to("deg") * pscale[1].to("deg")).value
+
+    computed_scale = compute_scale(wcs=wcs, fiducial=fiducial_world.value)
+
+    assert np.isclose(expected_scale, computed_scale)
+
+
+@pytest.mark.parametrize("datamodel_type", ["jwst", "roman"])
+def test_wcs_from_footprints(datamodel_type):
+    shape = (3, 3) * u.pix
+    fiducial_world = (10, 0) * u.deg
+    pscale = (0.1, 0.1) * u.arcsec
```

Comment:
What is `pscale`? Sounds like a scale but in angular units? And then used as a shift?

Reply:
>I am not sure why quantities are chosen to construct the WCS. neither Roman nor JWST use quantities or units for their WCSs.

The units were being used just for the sake of clarity; I will remove them.

> What is pscale? Sounds like a scale but in angular units? And then used as a shift?

`pscale` is indeed the plate scale in arcsec/pixel. The "per pixel" part was omitted because it was causing issues when performing arithmetic operations with dimensionless variables. Again, I'll remove the unit usage altogether.

Reply:
Units were removed.